storage/tests/unit/test__helpers.py

```diff
@@ -62,6 +73,25 @@ def test_reload(self):
         # Make sure changes get reset by reload.
         self.assertEqual(derived._changes, set())
 
+    def test_reload_w_user_project(self):
+        user_project = 'user-project-123'
+        connection = _Connection({'foo': 'Foo'})
+        client = _Client(connection)
+        derived = self._derivedClass('/path', user_project)()
+        # Make sure changes is not a set, so we can observe a change.
+        derived._changes = object()
+        derived.reload(client=client)
+        self.assertEqual(derived._properties, {'foo': 'Foo'})
+        kw = connection._requested
+        self.assertEqual(len(kw), 1)
+        self.assertEqual(kw[0]['method'], 'GET')
+        self.assertEqual(kw[0]['path'], '/path')
+        self.assertEqual(
+            kw[0]['query_params'],
+            {'projection': 'noAcl', 'userProject': user_project})
```

Comment:
Why don't you just compare `kw[0]` to a dict? This just leaves me (as the reader) wondering if there are unconsidered fields in `kw[0]`.